src/core/coap/coap.cpp

```diff
@@ -456,7 +443,8 @@ otError Coap::SendCopy(const Message &aMessage, const Ip6::MessageInfo &aMessage
     Message *messageCopy = NULL;
 
     // Create a message copy for lower layers.
-    VerifyOrExit((messageCopy = aMessage.Clone(aMessage.GetLength() - sizeof(CoapMetadata))) != NULL,
+    VerifyOrExit((messageCopy = static_cast<Message *>(aMessage.Clone(aMessage.GetLength() - sizeof(CoapMetadata)))) !=
```

Comment:
Is this `static_cast` necessary with the introduction of `ot::Coap::Message::Clone()` methods?